test.sh

```diff
@@ -0,0 +1,160 @@
+  DOCKER_MACHINE_ACTIVE=$(docker-machine active)
+  DOCKER_IP=$(docker-machine ip $DOCKER_MACHINE_ACTIVE)
+fi
+
+## Log settings
+export DOCKER_IP
+debug "DOCKER_IP=$DOCKER_IP"
+
+export FIXTURE_PREFIX=${FIXTURE_PREFIX:-"cbfix_"}
+debug "FIXTURE_PREFIX=$FIXTURE_PREFIX"
+
+debug "ROOT_DIR=$ROOT_DIR"
+debug "FIXTURE_DIR=$FIXTURE_DIR"
+debug "TESTS_DIR=$TESTS_DIR"
+
+## Main Code
+
+create_test_fixtures() {
+  banner "Create Test Fixtures"
+  if [ ! -f "$DIR/build/containerbuddy" ]; then die "Containerbuddy not built. Did you make?"; fi
+  find $FIXTURE_DIR -maxdepth 1 -mindepth 1 -type d | \
+    sort | \
+    while read FDIR; do
+      FNAME="${FIXTURE_PREFIX}$(basename $FDIR)"
+      rm -rf $FDIR/build
+      cp -r $DIR/build $FDIR/build
+      cd $FDIR
+      if [ -z "$(docker images -q $FNAME)" ]; then
+        log "Create fixture $FNAME"
+        docker build -t $FNAME --no-cache=false . 2>&1 >> ${DEBUG_LOG}
+        rm -rf $FDIR/build
+      else
+        log "Skipping Fixture $FNAME ... already exists"
+      fi
+  done
+}
+
+destroy_test_fixtures() {
+    banner "Destroying Test Fixtures"
+    find $FIXTURE_DIR -maxdepth 1 -mindepth 1 -type d |\
+      while read FDIR; do
+        FNAME="${FIXTURE_PREFIX}$(basename $FDIR)"
+        if [ -n "$(docker images -q $FNAME)" ]; then
+          log "Remove Fixture fixture $FNAME"
+          docker ps -q --filter "ancestor=${FNAME}" | while read CONTAINER_ID; do
+            log " - Killing $CONTAINER_ID"
+            docker rm --force $CONTAINER_ID >> ${DEBUG_LOG} 2>&1
+          done
+          docker rmi --force $FNAME >> ${DEBUG_LOG} 2>&1
+          rm -rf $FDIR/build
+        fi
+    done
+}
+
+build_test_compose() {
+  if [ -r "$COMPOSE_FILE" ]; then
+    docker-compose build >> ${DEBUG_LOG} 2>&1
+  fi
+}
+
+destroy_test_compose() {
+  if [ -r "$COMPOSE_FILE" ]; then
+    docker-compose kill >> ${DEBUG_LOG} 2>&1
+    docker-compose rm -f >> ${DEBUG_LOG} 2>&1
+  fi
+}
+
+run_tests() {
+  banner "Running Integration Tests"
+  FAILED=$TESTS_DIR/failed.log
+  rm -f $FAILED
+  for TDIR in $(find $TESTS_DIR -maxdepth 1 -mindepth 1 -type d ); do
+    if [ ! -r "$TDIR/run.sh" ]; then
+      log "WARN - No run.sh found in $TDIR"
+    else
+      cd $TDIR
+      export COMPOSE_PROJECT_NAME="$(basename $TDIR)"
+      export COMPOSE_FILE="./docker-compose.yml"
+      log "TEST: $COMPOSE_PROJECT_NAME"
+      build_test_compose
+      chmod 755 "./run.sh"
+      if ./run.sh; then
+        log "PASS: $COMPOSE_PROJECT_NAME"
+      else
+        log "FAIL: $COMPOSE_PROJECT_NAME"
+        echo $COMPOSE_PROJECT_NAME >> "$FAILED"
+      fi
+      destroy_test_compose
+    fi
+  done
+  [ ! -r "$FAILED" ]
+}
+
+clean_tests() {
+  find $TESTS_DIR -maxdepth 1 -mindepth 1 -type d |\
+    while read TDIR; do
+      cd $TDIR
+      export COMPOSE_PROJECT_NAME="$(basename $TDIR)"
+      export COMPOSE_FILE="./docker-compose.yml"
+      destroy_test_compose
+    done
+}
+
+case ${1:-"test"} in
```

Comment:
Maybe we should have the option to run single tests here? I could imagine a case where if a developer is trying to fix a broken test that having to wait for the whole rig to build would be painful. That does mean that:

> *Note*: Since fixtures are created in alpha order, they can have FROM directives for previously created images

probably wouldn't work. But maybe that's a good thing?